### PR TITLE
move the HTML extension to an external Composer package

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: extra/html-extension/composer.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            "role": "Lead Developer"
+        }
+    ],
+    "require": {
+        "php": "^7.1.3",
+        "twig/twig": "^2.4|^3.0"
+    },
+    "require-dev": {
+        "symfony/mime": "^4.3",
+        "symfony/phpunit-bridge": "^4.4@dev"
```
Comment (Tobion) [inline extra/html-extension/composer.json:22]: I think this should be moved to "require" or?

Comment (fabpot) [reply extra/html-extension/composer.json:22]: The mime component is "only" used for 1 filter (not the other) and only if you don't pass the content-type. I hesitated on this one. Might be "better" to always require it.

Comment (Tobion) [reply extra/html-extension/composer.json:22]: I would try to get rid of optional dependencies.

Comment (Tobion) [reply extra/html-extension/composer.json:22]: And installing it when it's not needed does seem better than not installing it, when it is needed.

**Comment by Tobion** [comment]
I don't know if you planned this already: With the extra optional extensions, it would be cool to enable extensions automatically when they are installed. So similar to symfony framework that enables stuff based on the existence. So we could have an extension in core that automatically installes other extra extensions when they exist.

**Comment by fabpot** [comment]
That's already the case. Have a look at the `extra/twig-extra-bundle` directory. It automatically register the extensions when they are installed and even better, they hint you when trying to use a filter/function that does not exist but is part of an extra extension.

**Pull request merged**